### ISSUE TITLE
Why the complex dependencies?

### ISSUE THREAD

**Issue by JayAndCatchFire** [pedestal/pedestal#533]
Hey, I'm a fan of the project and I wanted to give back by helping update some of the outdated dependencies.

I started updating dependencies and I noticed some oddities.  For example, `potemkin` is used by `clj-http` as a dev dependency in `pedestal.service`, but it's excluded there.  It's a dependency of `pedestal.immutant` even though it's never actually used in the code anywhere.

Another example is that `javax.servlet/javax.servlet-api` is a dependency of `pedestal.immutant` but it's never used anywhere in that project.

Another example is that `crypto-equality` is excluded from the dependency `ring/ring-core` in `pedestal.service` and then included on its own.  I figured this was probably to get a specific version of `crypto-equality`.  However, the version of `ring/ring-core` (1.5.1, which is out of date incidentally) doesn't depend on `crypto-equality` anymore.

So are the dependencies of the pedestal packages in need of some pruning or is there some method to the madness?  Because if it's the former I'd be more than happy to volunteer my time weeding some of this stuff out.

**Comment by ohpauleez** [comment]
Hi @JayAndCatchFire -- thanks for reaching out and volunteering to step up!

I usually go through and update the dependencies just prior to a release, with `pedantic? :abort` on, and patch up dependencies as I go.  Usually dev dependencies are only updated as needed.
All platforms (immutant, jetty, tomcat, etc.) that connect to Pedestal via a servlet interface, need to include the servlet API (which is a provided dep in Pedestal Service).

If you find explicit exclusions and those exclusions appear as deps in a Pedestal module (like Ring and crypto-random in Pedestal Service), it's most likely:
 1. At some point in history, the sub dependencies were updated with bugfixes that were important to Pedestal
 2. Two dependencies depended on the sub dependencies and we needed to fix the classpath.

Once one of those situations happens, I rarely go back to check if it can later be undone (since functionally speaking, it doesn't matter for the Pedestal projects), but I'll happily take any help cleaning it up again!

So in summary, feel free to clean up the deps however you like, but I wouldn't bother with the dev dependencies.  Make sure you have `:pendatic? :abort` on in all of the Pedestal projects, make exclusions explicit, and put all "clean up" deps at the bottom/end of the deps vector.  Thanks!

**Comment by JayAndCatchFire** [comment]
Thanks!  Expect a slew of PRs that update dependencies, then.  I'm probably going to update related stuff as separate PRs so I can ensure I'm not breaking any tests.  Y'know, until I do start breaking stuff.  But those will be their own PRs.

Can't wait to start contributing!
